src/arweave_address.ts

```diff
@@ -7,15 +7,19 @@ export class ArweaveAddress {
 		}
 	}
 
-	equalsAddress(other: ArweaveAddress): boolean {
-		return this.address === other.address;
+	[Symbol.toPrimitive](hint: 'number' | 'string' | 'default'): string {
+		if (hint === 'number') {
+			throw new Error('Address is a string!!!!');
```

Comment:
When would this error be helpful for us? If this isn't helping us at compile time, when is it helping us? Also this error message is confusing because the reader won't be sure if whether:
a) address is a number when it shouldn't be
b) address should be a number (it should not)

Reply:
(b) is actually a funny case because there could be rare cases where a numeric address is valid! 😱 But maybe not in code? Unsure.

Reply:
> maybe not in code?
I see your test example now:
```
                 it('throws if casted to a number', () => {
			const address = new ArweaveAddress('g1hzNXVbh2M6LMQSUYp7HgkgxdadYqYEfw-HAajlms0');
			expect(() => +address).to.throw();
		});
```
I like the protection offered by your change.

Reply:
I'm 80% sure we only need `toString()` for all of the safe types. If we remove `toPrimitive` and `valueOf`, I would say `+address === NaN` is the only difference

Reply:
Unless we do want that type of protection for numbers we can remove all of those functions

Reply:
Regarding #123 and #129